src/test/java/io/debezium/operator/dependent/ServiceAccountDependentTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.operator.dependent;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.debezium.operator.DebeziumServer;
+import io.debezium.operator.DebeziumServerReconciler;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.api.model.ServiceAccount;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.ReconcilerUtils;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.quarkus.kubernetes.client.runtime.KubernetesClientUtils;
+
+public class ServiceAccountDependentTest {
+
+    public static final String dsName = "test-ds";
+    private static final String managedSaName = "test-ds-sa";
+    KubernetesClient client = KubernetesClientUtils.createClient();
+    DebeziumServer debeziumServer;
+
+    private Context<DebeziumServer> context;
+
+    @BeforeEach
+    void before() {
+        debeziumServer = ReconcilerUtils.loadYaml(
+                DebeziumServer.class, DebeziumServerReconciler.class, "ds-postgres-minimal.yml");
+        final var metadata = new ObjectMetaBuilder()
+                .withName(dsName)
+                .withNamespace(client.getNamespace())
+                .build();
+        debeziumServer.setMetadata(metadata);
+    }
+
+    @AfterEach
+    void after() {
+        client.resource(debeziumServer).delete();
+    }
+
+    @Test
+    void shouldReturnServiceAccount() {
+        final var saName = "test-ds-service-account";
+
+        debeziumServer.getSpec().getRuntime().setServiceAccount(saName);
+        String sa = ServiceAccountDependent.serviceAccountNameFor(debeziumServer);
+
+        assertThat(sa).isEqualTo(saName);
+    }
+
+    @Test
+    void shouldReturnManagerServiceAccount() {
+        String sa = ServiceAccountDependent.serviceAccountNameFor(debeziumServer);
+        assertThat(sa).isEqualTo(managedSaName);
+    }
+
+    @Test
+    void shouldDesiredServiceAccount() {
+        ServiceAccountDependent serviceAccountDependent = new ServiceAccountDependent();
+        ServiceAccount serviceAccount = serviceAccountDependent.desired(debeziumServer, context);
+
+        assertThat(serviceAccount.getMetadata().getName()).isEqualTo(managedSaName);
+        assertThat(serviceAccount.getMetadata().getNamespace()).isEqualTo("default");
```

Comment:
`default` is the default so IMHO it might be best to change the namespace first and tests with that.